src/_guides/language/effective-dart/usage.md

```diff
@@ -441,33 +444,110 @@ var reciprocals = stuff.map((n) => 1 / n).cast<double>();
+  var list = []; // List<dynamic>.
+  if (a != null) list.add(a);
+  if (b != null) list.add(b);
+  return list.cast<int>();
+}
+{% endprettify %}
+
+#### Cast the elements on access
+
+If you immediately iterate over the collection, cast each element inside the
+iteration:
 
-*   If you immediately iterate over the collection, cast each element inside the
-    iteration.
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/usage_good.dart (cast-iterate)" replace="/int n/[!int!] n/g"?>
+{% prettify dart %}
+int sum(List<Object> objects) {
+  var result = 0;
+  // We happen to know the list only contains ints.
+  for ([!int!] n in objects) {
+    result += n;
+  }
 
-*   If you'll eventually access most of the elements in the collection, and
-    you don't need the object to be backed by the original live object, convert
-    it using `List.from()`.
+  return result;
+}
+{% endprettify %}
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (cast-iterate)"?>
+{% prettify dart %}
+int sum(List<Object> objects) {
+  var result = 0;
+  // We happen to know the list only contains ints.
+  for (var n in objects.cast<int>()) {
+    result += n;
+  }
 
-    The `cast()` and `retype()` methods return lazy collections that check the
-    element type on *every operation*. If you perform only a few operations on
-    only a few elements, that laziness can be good. But in many cases, the
-    overhead of lazy validation and of wrapping outweighs the benefits.
+  return result;
+}
+{% endprettify %}
 
-These alternatives don't always work, of course, and sometimes `cast()` or
-`retype()` is the right answer. But consider those methods to be a little risky
-and undesirable&mdash;they can be slow and may produce errors at runtime if you
-aren't careful.
+#### Eagerly cast using `List.from()`
+
+If you'll eventually access most of the elements in the collection, and you
+don't need the object to be backed by the original live object, convert it using
+`List.from()`:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/usage_good.dart (cast-from)"?>
+{% prettify dart %}
+int median(List<Object> objects) {
+  // We happen to know the list only contains ints.
+  var ints = objects.cast<int>();
+  ints.sort();
+  return ints[ints.length ~/ 2];
+}
+{% endprettify %}
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (cast-from)"?>
+{% prettify dart %}
+int median(List<Object> objects) {
+  // We happen to know the list only contains ints.
+  var ints = new List<int>.from(objects);
+  ints.sort();
+  return ints[ints.length ~/ 2];
+}
+{% endprettify %}
+
+The `cast()` method returns a lazy collection that checks the element type on
+*every operation*. If you perform only a few operations on only a few elements,
+that laziness can be good. But in many cases, the overhead of lazy validation
+and of wrapping outweighs the benefits.
+
+These alternatives don't always work, of course, and sometimes `cast()` is the
```

Comment:
Move this paragraph up, to just after the bulleted list.

Reply:
With the examples split out from the list, I like this paragraph at the end now because I feel like it summarizes the whole thing.

Reply:
It does, and yet no one might see it. With the headings, it looks like it only applies to the last section, and the scope of "These alternatives" is unclear.

Reply:
But if we remove the headings, it can stay!! (My soon-to-be-sent review has a suggestion for that.)